### PR TITLE
chore(ts): Convert utils/{withP,p}rofiler

### PR DESCRIPTION
**Pull request by evanpurkhiser**
Refactors to remove the unneded function call

### CODE REVIEW

File: src/sentry/static/sentry/app/utils/withProfiler.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+import {Integrations} from '@sentry/apm';
+
+import getDisplayName from 'app/utils/getDisplayName';
+
+type InjectedProps = {
+  finishProfile: () => void;
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/utils/withProfiler.tsx]: I tried to figure out a way to make this type not _required_ to be on the wrapped components type, but not sure if that's possible?

File: src/sentry/static/sentry/app/utils/withProfiler.tsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        return;
+      }
+
+      Integrations.Tracing.popActivity(this.activity);
+      this.activity = null;
+    };
+
+    render() {
+      return (
+        <WrappedComponent {...(this.props as P)} finishProfile={this.finishProfile} />
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/utils/withProfiler.tsx]: Maybe we could make the `finishProfile` an optional prop with a config or something

File: src/sentry/static/sentry/app/utils/withProfiler.tsx (hunk 1 above)
Comment (markstory) [reply src/sentry/static/sentry/app/utils/withProfiler.tsx]: I don't think it is possible, as you can't easily inspect the wrapped component's props. Without passing the finishProfile callback this HOC does very little though so I think that `finishProfile` is a reasonable contract to have.

**Review by markstory — APPROVED** [review]

File: src/sentry/static/sentry/app/views/userFeedback/index.tsx [hunk 3]
```diff
@@ -30,13 +29,10 @@ type State = AsyncView['state'] & {
 
 type Props = RouteComponentProps<{orgId: string}, {}> & {
   organization: Organization;
+  finishProfile: () => void;
```
Comment (billyvg) [inline src/sentry/static/sentry/app/views/userFeedback/index.tsx]: Is there a way to refer to `InjectedProps` via `withProfile`?

**Review by billyvg — APPROVED** [review]

File: src/sentry/static/sentry/app/views/userFeedback/index.tsx (hunk 3 above)
Comment (EvanPurkhiser) [reply src/sentry/static/sentry/app/views/userFeedback/index.tsx]: There might be a way to infer it from the HoC, but I think the signature is simple enough to do this

**Pull request merged**